topologytest/couchbase_lite_mock_peer_test.go

```diff
@@ -135,20 +145,36 @@ func (p *CouchbaseLiteMockPeer) WaitForDocVersion(dsName sgbucket.DataStoreName,
 }
 
 // WaitForDeletion waits for a document to be deleted. This document must be a tombstone. The test will fail if the document still exists after 20s.
-func (p *CouchbaseLiteMockPeer) WaitForDeletion(_ sgbucket.DataStoreName, _ string) {
-	require.Fail(p.TB(), "WaitForDeletion not yet implemented CBG-4257")
+func (p *CouchbaseLiteMockPeer) WaitForDeletion(dsName sgbucket.DataStoreName, docID string) {
+	client := p.getSingleSGBlipClient().CollectionClient(dsName)
+	require.EventuallyWithT(p.TB(), func(c *assert.CollectT) {
+		isTombstone, err := client.IsTombstoned(docID)
+		require.NoError(c, err)
+		require.True(c, isTombstone, "expected docID %s on peer %s to be deleted", docID, p)
+	}, totalWaitTime, pollInterval)
 }
 
 // WaitForTombstoneVersion waits for a document to reach a specific version, this must be a tombstone. The test will fail if the document does not reach the expected version in 20s.
-func (p *CouchbaseLiteMockPeer) WaitForTombstoneVersion(_ sgbucket.DataStoreName, _ string, _ DocMetadata) {
-	require.Fail(p.TB(), "WaitForTombstoneVersion not yet implemented CBG-4257")
+func (p *CouchbaseLiteMockPeer) WaitForTombstoneVersion(dsName sgbucket.DataStoreName, docID string, expected DocMetadata) {
+	client := p.getSingleSGBlipClient().CollectionClient(dsName)
+	expectedVersion := db.DocVersion{CV: expected.CV(p.TB())}
+	require.EventuallyWithT(p.TB(), func(c *assert.CollectT) {
+		isTombstone, err := client.IsVersionTombstone(docID, expectedVersion)
+		require.NoError(c, err)
+		require.True(c, isTombstone, "expected docID %s on peer %s to be deleted", docID, p)
+	}, totalWaitTime, pollInterval)
 }
 
 // RequireDocNotFound asserts that a document does not exist on the peer.
-func (p *CouchbaseLiteMockPeer) RequireDocNotFound(sgbucket.DataStoreName, string) {
-	// not implemented yet in blip client tester
-	// _, err := p.btcRunner.GetDoc(p.btc.id, docID)
-	// base.RequireDocNotFoundError(p.btcRunner.TB(), err)
+func (p *CouchbaseLiteMockPeer) RequireDocNotFound(dsName sgbucket.DataStoreName, docID string) {
+	client := p.getSingleSGBlipClient().CollectionClient(dsName)
+	isTombstone, err := client.IsTombstoned(docID)
```

Comment:
See above, the thing that is really helpful is to show the body + version if this function fails.

This function isn't actually used in tests IIRC.